browser/brave_tab_helpers.cc

```diff
@@ -132,9 +135,14 @@ void AttachTabHelpers(content::WebContents* web_contents) {
       web_contents);
 
   brave_ads::AdsTabHelper::CreateForWebContents(web_contents);
-
   brave_ads::SearchResultAdTabHelper::MaybeCreateForWebContents(web_contents);
 
+  if (!web_contents->GetBrowserContext()->IsOffTheRecord()) {
```

Comment:
this is actually backwards and we should avoid following this pattern. You should use `MaybeCreateForWebContents` and put the OTR/feature flag check inside of that